formats/json-tests/commonTest/src/kotlinx/serialization/json/serializers/JsonPrimitiveSerializerTest.kt

```diff
@@ -201,4 +202,17 @@ class JsonPrimitiveSerializerTest : JsonTestBase() {
             assertUnsignedNumberEncoding(expected, actual, JsonPrimitive(actual))
         }
     }
+
+    @Serializable
+    class OuterLong(val a: Long)
+
+    @Test
+    fun testRejectingIncorrectNumbers() = parametrizedTest { mode ->
+        checkSerializationException({
+            default.decodeFromString(OuterLong.serializer(), """{"a":"12:34:45"}""", mode)
+        }, {
+            if (mode == JsonTestingMode.TREE) assertContains(it, "Failed to parse literal '\"12:34:45\"' as a long value at element: \$.a")
```

Comment:
Should we also check exception type?

Reply:
`checkSerializationException` makes sure that only `SerializationException`s are accepted